### ISSUE TITLE
ng update to 12 is pulling in 13.0.0-next.1 

### ISSUE THREAD

**Issue by pauldaustin** [angular/angular-cli#21571]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

### Description

Old version 9

ng update --allow-dirty  @angular/core@12  @angular/cli@12  @angular/cdk@12  @angular/material@12

Looks like it has jumped to 13.0.0-next.1

errors

   Package "@angular-devkit/build-angular" has an incompatible peer dependency to "@angular/compiler-cli" (requires "^12.0.0 || ^12.2.0-next", would install "13.0.0-next.1")
                  Package "@angular/animations" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1", would install "12.2.1")
                  Package "@angular/common" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1", would install "12.2.1")
                  Package "@angular/animations" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1" (extended), would install "12.2.1").
                  Package "@angular/forms" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1", would install "12.2.1")
                  Package "@angular/platform-browser" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1", would install "12.2.1")
                  Package "@angular/platform-browser-dynamic" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1", would install "12.2.1")
                  Package "@angular/router" has an incompatible peer dependency to "@angular/core" (requires "13.0.0-next.1", would install "12.2.1")

## 🔥 Exception or Error

<pre><code>

</code></pre>

## 🌍 Your Environment

<pre><code>

</code></pre>

**Anything else relevant?**

**Comment by alan-agius4** [comment]
Can you please share you package.json before the update?

**Comment by pauldaustin** [comment]
```json
{
  "name": "xxxxx-ui",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "build-prod": "ng build --aot --baseHref=/pub/xxxxx/ --prod --preserve-symlinks --crossOrigin=anonymous --progress false",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^9.1.13",
    "@angular/cdk": "^9.2.4",
    "@angular/common": "^9.1.13",
    "@angular/compiler": "^9.1.13",
    "@angular/core": "^9.1.13",
    "@angular/forms": "^9.1.13",
    "@angular/material": "^9.2.4",
    "@angular/platform-browser": "^9.1.13",
    "@angular/platform-browser-dynamic": "^9.1.13",
    "@angular/router": "^9.1.13",
    "@fortawesome/fontawesome-free": "^5.15.4",
    "leaflet": "^1.7.1",
    "rxjs": "^6.6.7",
    "tslib": "^1.14.1",
    "zone.js": "^0.10.3"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^0.901.15",
    "@angular/cli": "^9.1.15",
    "@angular/compiler-cli": "^9.1.13",
    "@angular/language-service": "^9.1.13",
    "@types/jasmine": "^3.8.2",
    "@types/jasminewd2": "^2.0.10",
    "@types/node": "^12.20.19",
    "codelyzer": "^5.2.2",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "^5.0.2",
    "karma": "^5.2.3",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "^3.0.3",
    "karma-jasmine": "^3.3.1",
    "karma-jasmine-html-reporter": "^1.7.0",
    "protractor": "^7.0.0",
    "ts-node": "^8.10.2",
    "tslint": "^6.1.3",
    "typescript": "~3.8.0"
  }
}

```

**Comment by clydin** [comment]
This should now be corrected with the latest version of v12 (12.2.2).
If the issue is still being encountered, please let us know.
